cmd/controller/main.go

```diff
@@ -77,6 +79,8 @@ func main() {
 
 	logging.FromContext(ctx).Infof("Initializing with version %s", project.Version)
 	// Set up controller runtime controller
+	var recorder events.Recorder = &events.NoOpRecorder{}
```

Comment:
I did this specifically so that everything that uses `recorder` would have to use the `events.Recorder` interface and we couldn't accidentally use the real type.